### PR TITLE
Add docker-compose support

### PR DESCRIPTION
**Pull request by nmalaguti**
This makes it very easy to start up lazy blacksmith locally or in the
cloud using docker-compose.

You must specify a `SECRET_KEY` in a `.env` file in the same directory
as `docker-compose.yml`. You can also set additional configuration
options in the `.env` file such as `ESI_SECRET_KEY`, `ESI_CLIENT_ID`,
and `ESI_REDIRECT_DOMAIN`.

You can easily start lazy blacksmith with `docker-compose up --build`
and update the SDE with `docker-compose run init-db sde-import`.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    depends_on:
+      init-db:
+        condition: service_completed_successfully
+    restart: always
+
+  celery-beat:
+    build:
+      context: .
+      dockerfile: deploy/Dockerfile
+    hostname: web
```
Comment (Kyria) [inline docker-compose.yml:92]: i believe this is a typo also, as "web" is already used by uwsgi.

**Comment by Kyria** [comment]
other than the reviews, it's a good idea to provide a default docker-compose config (and to add a few things in the entrypoints), ty :)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    env_file: .env
+    environment: &service_environment
+      UWSGI_SOCKET_TYPE: ${UWSGI_SOCKET_TYPE:---http}
+      CELERY_BROKER: redis://redis:6379/0
+      CELERY_RESULT_BACKEND: redis://redis:6379/0
+      DB_URI: postgresql://lazyblacksmith:${POSTGRES_PASSWORD:-lazyblacksmith}@postgres/lazyblacksmith
+      SECRET_KEY: ${SECRET_KEY?SECRET_KEY must be set in your .env file}
+      ESI_SECRET_KEY: ${ESI_SECRET_KEY-}
+      ESI_CLIENT_ID: ${ESI_CLIENT_ID-}
+      ESI_REDIRECT_DOMAIN: ${ESI_REDIRECT_DOMAIN-}
```
[the remark this answers is not in the document]
Comment (nmalaguti) [reply docker-compose.yml:45]: From the docker-compose docs: https://docs.docker.com/compose/compose-file/#interpolation

> `${VARIABLE-default}` evaluates to default only if VARIABLE is unset in the environment.

It allows you to explicitly pass an empty string.

File: docker-compose.yml (hunk 1 above)
Comment (nmalaguti) [reply docker-compose.yml]: you are right! Good catch.

**Comment by nmalaguti** [comment]
@Kyria ready for re-review

**Comment by Kyria** [comment]
sadly didn't have time to have an extensive look at it tonight, i'll do it tomorrow. 
So far, looks good to me, just want to double check again before merging :)

**Pull request merged**